src/main/java/com/capitalone/dashboard/service/CodeQualityServiceImpl.java

```diff
@@ -68,6 +69,33 @@ public CodeQualityServiceImpl(CodeQualityRepository codeQualityRepository,
         return searchType(request);
     }
 
+    @Override
+    public Iterable<CodeQuality> getAllSecurityScans(CodeQualityRequest request) {
+
+        Component component = componentRepository.findOne(request.getComponentId());
+
+        Iterable<CollectorItem> collectorItems = component.getCollectorItems(request.getType().collectorType());
+
+        ArrayList<CodeQuality> codeQualities = new ArrayList<CodeQuality>();
+
+        collectorItems.forEach(item -> {
+            CodeQuality tempCQ = codeQualityRepository.findTop1ByCollectorItemIdOrderByTimestampDesc(item.getId());
+            String name = (String) item.getOptions().get("projectName");
+            String reportUrl = (String) item.getOptions().get("reportUrl");
+
+            // If name is null, use the first part of the description
+            if(name.equals("") || name.equals(null)) {
+                name = item.getDescription().split(":")[0];
+            }
+
+            tempCQ.setName(name);
```

Comment:
@tatlax3636 plz add null check for tempCQ.

Reply:
Added